silk/unit_tests/test_otns.py

```diff
@@ -0,0 +1,450 @@
+
+  def testUpdateRLOC16(self):
+    """Test updating node RLOC16.
+
+    Also tests updating before the OTNS manager subscribes to the node.
+    """
+    ns = self.ns
+    manager = self.manager
+
+    device_id = random.randint(1, 10)
+    device_rloc16 = random.getrandbits(16)
+    device = MockThreadDevBoard("device", device_id)
+    wpantund_process = MockWpantundProcess()
+    device.wpantund_process = wpantund_process
+
+    manager.add_node(device)
+    ns.go(0.1)
+
+    original_rloc16 = ns.nodes()[device_id]["rloc16"]
+
+    wpantund_process.emit_status(f"rloc16={device_rloc16}")
+    ns.go(0.1)
+
+    self.assertEqual(ns.nodes()[device_id]["rloc16"], original_rloc16)
+
+    manager.subscribe_to_node(device)
+    wpantund_process.emit_status(f"rloc16={device_rloc16}")
+    ns.go(0.1)
+
+    self.assertEqual(ns.nodes()[device_id]["rloc16"], device_rloc16)
+
+  def testFormPartition(self):
+    """Test forming a partition.
+    """
+    ns = self.ns
+    manager = self.manager
+
+    device_1_id = random.randint(1, 10)
+    device_1_parid = random.getrandbits(16)
+    device_1 = MockThreadDevBoard("device_1", device_1_id)
+    wpantund_process_1 = MockWpantundProcess()
+    device_1.wpantund_process = wpantund_process_1
+
+    device_2_id = random.randint(11, 20)
+    device_2_parid = random.getrandbits(16)
+    device_2 = MockThreadDevBoard("device_2", device_2_id)
+    wpantund_process_2 = MockWpantundProcess()
+    device_2.wpantund_process = wpantund_process_2
+
+    manager.add_node(device_1)
+    manager.add_node(device_2)
+
+    manager.subscribe_to_node(device_1)
+    manager.subscribe_to_node(device_2)
+
+    wpantund_process_1.emit_status(f"parid={device_1_parid:08x}")
+    wpantund_process_2.emit_status(f"parid={device_2_parid:08x}")
+    ns.go(0.1)
+
+    partitions_info = ns.partitions()
+    self.assertEqual(len(partitions_info), 2)
+    self.assertEqual(len(partitions_info[device_1_parid]), 1)
+    self.assertEqual(len(partitions_info[device_2_parid]), 1)
+    self.assertEqual(partitions_info[device_1_parid][0], device_1_id)
+    self.assertEqual(partitions_info[device_2_parid][0], device_2_id)
+
+    wpantund_process_2.emit_status(f"parid={device_1_parid:08x}")
+    ns.go(0.1)
+
+    partitions_info = ns.partitions()
+    self.assertEqual(len(partitions_info), 1)
+    self.assertEqual(len(partitions_info[device_1_parid]), 2)
+    self.assertTrue(device_1_id in partitions_info[device_1_parid])
+    self.assertTrue(device_2_id in partitions_info[device_1_parid])
+
+    wpantund_process_2.emit_status(f"parid={device_2_parid:08x}")
+    ns.go(0.1)
+
+    partitions_info = ns.partitions()
+    self.assertEqual(len(partitions_info), 2)
+    self.assertEqual(len(partitions_info[device_1_parid]), 1)
+    self.assertEqual(len(partitions_info[device_2_parid]), 1)
+    self.assertEqual(partitions_info[device_1_parid][0], device_1_id)
+    self.assertEqual(partitions_info[device_2_parid][0], device_2_id)
+
```

Comment:
Test cild table and router table update ?

Reply:
The set of commands supported by pyOTNS is limited to the ones supported by OTNS CLI (https://github.com/openthread/ot-ns/blob/master/cli/README.md). It doesn't support querying child and router tables yet. I plan to add that test to the up coming unit test but I'm still thinking of the best way to test that.
Another path to take is to work on adding that support to OTNS CLI first.

Reply:
TODO added at the end.